common.py

```diff
@@ -1,7 +1,7 @@
 import jinja2
 from dataclasses import dataclass, field
 from enum import Enum
-from typing import List
+from typing import List, Optional, Any, Dict, List, Union, Type, TypeVar, cast
```

Comment:
If this depends on Python 3.9+, then you don't need the deprecated `typing.{List,Dict,Tuple}` type but can use `{list,dict,tuple}` directly. See https://docs.python.org/3/library/typing.html#typing.List

Depending on Python 3.9+ seems reasonable since,
- it's in RHEL9
- CDA also depends on Python 3.11+

btw, `List` is imported twice. I would think that a linter complains about that too...

Reply:
(it's fine, this can be changed in a follow-up commit too. It it causes extra work to rebase)